command/input/handler.go

```diff
@@ -205,7 +207,18 @@ func (e *Handler) HandleEvent(focused input.Editor, ev gxui.KeyboardEvent) {
 			}
 			edits = append(edits, edit)
 		}
-		e.Apply(focused, edits...)
+		if len(selections) == 1 && selections[0].Start() >= 0 {
+			pos := selections[0].Start()
+			e.Apply(focused, edits...)
+			if selections[0].Length() != 0 {
+				editor.SetScrollOffset(editor.StartOffset())
+			} else {
+				lineIndex := math.Max(editor.LineIndex(pos)-3, 0)
+				editor.ScrollToLine(lineIndex)
+			}
```

Comment:
I would like to see the scrolling logic in an `op`, similar to [`caret.Mover`](command/caret/caret.go), and a hook to scroll when text changes, like [`caret.OnEdit`](command/caret/on_edit.go).  It should also probably trigger when the caret moves.

The reason for this is that it allows the `input` package to exclusively focus on handling user input, and it allows other commands or plugins that might need to scroll the screen to just load the `scroll` op.  Plus, if anything needs to trigger when the screen is scrolled, we can always add hooks to the scroll op.